SS2-Project/Assets/Starstorm2/ScriptableObjects/SS2AssetCollections.cs

```diff
@@ -0,0 +1,99 @@
+        public GameObject bodyPrefab;
+        public GameObject masterPrefab;
+
+        public void OnValidate()
+        {
+            //todo: ensure field values are also contained in AssetCollection.assets
+        }
+    }
+    [CreateAssetMenu(fileName = "SurvivorAssetCollection", menuName = "Starstorm2/AssetCollections/SurvivorAssetCollection")]
+    public class SurvivorAssetCollection : BodyAssetCollection
+    {
+        public SurvivorDef survivorDef;
+    }
+    [CreateAssetMenu(fileName = "MonsterAssetCollection", menuName = "Starstorm2/AssetCollections/MonsterAssetCollection")]
+    public class MonsterAssetCollection : BodyAssetCollection
+    {
+        public NullableRef<MonsterCardProvider> monsterCardProvider;
+    }
+    [CreateAssetMenu(fileName = "ArtifactAssetCollection", menuName = "Starstorm2/AssetCollections/ArtifactAssetCollection")]
+    public class ArtifactAssetCollection : AssetCollection
+    {
+        public NullableRef<ArtifactCode> artifactCode;
+        public ArtifactDef artifactDef;
+    }
+    [CreateAssetMenu(fileName = "EquipmentAssetCollection", menuName = "Starstorm2/AssetCollections/EquipmentAssetCollection")]
+    public class EquipmentAssetCollection : AssetCollection
+    {
+        public GameObject[] itemDisplayPrefabs;
+        public EquipmentDef equipmentDef;
+    }
+    [CreateAssetMenu(fileName = "EliteAssetCollection", menuName = "Starstorm2/AssetCollections/EliteAssetCollection")]
+    public class EliteAssetCollection : EquipmentAssetCollection
+    {
+        public EliteDef[] eliteDefs;
+    }
+    [CreateAssetMenu(fileName = "InteractableAssetCollection", menuName = "Starstorm2/AssetCollections/InteractableAssetCollection")]
+    public class InteractableAssetCollection : AssetCollection
+    {
+        public GameObject interactablePrefab;
+        public InteractableCardProvider interactableCardProvider;
+    }
+    [CreateAssetMenu(fileName = "ItemAssetCollection", menuName = "Starstorm2/AssetCollections/ItemAssetCollection")]
+    public class ItemAssetCollection : AssetCollection
+    {
+        public GameObject[] itemDisplayPrefabs;
+        public EquipmentDef itemDef;
+    }
+    [CreateAssetMenu(fileName = "ItemTierAssetCollection", menuName = "Starstorm2/AssetCollections/ItemTierAssetCollection")]
+    public class ItemTierAssetCollection : AssetCollection
+    {
+
+    }
+
+
+    [CustomEditor(typeof(AssetCollection))]
+    public class AssetCollectionEditor : Editor
+    {
+        public override void OnInspectorGUI()
+        {
+            DrawDefaultInspector();
+            EditorGUILayout.BeginVertical("box");
+
+            if (GUILayout.Button("Add Selected Objects"))
+            {
+                AddSelection();
+            }
+            EditorGUILayout.EndVertical();
+        }
+
+        private void AddSelection()
+        {
+            //todo: add folder selection and actually sort objects in selection (reuse contentutils.addcontentfromassetcollection)
```

Comment:
You could look into how Thunderkit collects all the assets inside a folder + any dependency assets.